### PR TITLE
feat(ktable): support for conditional events

### PR DESCRIPTION
**Pull request by kaiarrowood**
# Summary

Backport https://github.com/Kong/kongponents/pull/637

Edit: we may end up not backporting this after all

## Vue 3
Better handling for condition events

**Did you create a corresponding PR to add this change to the `beta` branch? (required)**

- [ ] **Yes**, here is a link to the PR: `[link to beta branch PR]`
- [ ] **No**, I did not create a PR for the `beta` branch and I have added the `port to beta branch` tag to this PR.
- [ ] **No**, because `[type your reasons]`

If you have questions, tag `@adamdehaven` or `@kaiarrowood`.

---

## PR Checklist

- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ module.exports = {
     'object-property-newline': 'error',
     'import/no-named-default': 0,
     'object-curly-spacing': 0,
+    'no-extra-boolean-cast': 0,
```
Comment (adamdehaven) [inline .eslintrc.js:25]: question: Was this intentional?

Comment (kaiarrowood) [reply .eslintrc.js:25]: Yes, it was mad about the `!!` in the conditional

File: docs/components/table.md [hunk 2]
```diff
@@ -599,20 +599,35 @@ export default {
 </script>
 ```
 
-## Events
+Bind DOM [events](https://developer.mozilla.org/en-US/docs/Web/Events) to various parts of the table. We support events on both table rows and cells, but must be careful with clickable content in rows when row click is enabled. You can also add logic to check for `metakey` to support cmd/ctrl when clicking. Examples highlighted below.
```
Comment (adamdehaven) [inline docs/components/table.md:602]: It looks like you accidentally deleted the `## Events` header

File: docs/components/table.md [hunk 3]
```diff
@@ -632,10 +647,12 @@ To avoid firing row clicks by accident, the row click handler ignores events com
 </KTable>
 
 ```html
+<KInputSwitch v-model="enableRowClick" :label="enableRowClick ? 'Row clicks enabled' : 'Row clicks disabled'" />
```
Comment (adamdehaven) [inline docs/components/table.md:650]: These switches aren't working -- the row clicks are not firing

**Review by adamdehaven — CHANGES_REQUESTED** [review]

**Pull request closed without merging**